docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+with default HTTP configuration on the default port, 8080.
+
+When a load balancer accepts a TLS encrypted connection, it places the following
+string in the HTML header of connection attempts:
+
+.. code-block:: text
+
+    X-Forwarded-Proto: https
+
+This tells the Trino coordinator to process the connection as if a TLS
+connection has already been successfully negotiated for it. This is why you do
+not need to configure ``http-server.https.enabled=true`` for a coordinator
+behind a load balancer.
+
+However, to enable processing of such forwarded headers, the server's
+:ref:`config properties file <config_properties>` *must* include the following:
+
+.. code-block:: text
+
+  http-server.process-forwarded=true
+
+This completes any necessary configuration for using HTTPS with a load balancer.
+Client tools can access Trino with the URL exposed by the load balancer.
+
+.. _https-secure-directly:
+
+Secure Trino directly
+----------------------
+
+Instead of the preferred mechanism of using an :ref:`external load balancer
+<https-load-balancer>`, you can secure the Trino coordinator itself. This
+requires you to obtain and install a TLS :ref:`certificate <glossCert>`, and
+configure Trino to use it for any client connections.
+
+Add a TLS certificate
+^^^^^^^^^^^^^^^^^^^^^
+
+Obtain a TLS certificate file for use with your Trino server. Consider the
+following types of certificates:
+
+* **Globally trusted certificates** — A certificate that is automatically
+  trusted by browsers and clients such as ``curl``. This is the easiest type to
+  use because you do not need to configure clients. Obtain a certificate of this
+  type from:
+
+  *  A commercial certificate vendor
+  *  Your cloud infrastructure provider
+  *  A domain name registrar, such as Verisign or GoDaddy
+  *  A free certificate generator, such as
+     `letsencrypt.org <https://letsencrypt.org/>`_ or
+     `sslforfree.com <https://www.sslforfree.com/>`_
+
+* **Corporate trusted certificates** — A certificate trusted by browsers and
+  clients in your organization. Typically, a site's IT department runs a local
+  :ref:`certificate authority <glossCA>` and preconfigures clients and servers
+  to trust this CA.
+
+* **Generated self-signed certificates** — A certificate generated just for
+  Trino that is not automatically trusted by any client. Before using, make sure
+  you understand the :ref:`limitations of self-signed certificates
+  <self_signed_limits>`.
+
+The most convenient option is to purchase a PEM certificate from a Certificate
+Authority that is already listed in the Java truststore. This case requires no
+extra configuration for clients to *just work* when connecting to your server.
+
+Keys, certificates, and containers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Trino recognizes public and private :ref:`keys <glossKey>` as specified in the
+PKCS #1 standard, and recognizes X.509 :ref:`certificates <glossCert>`. Trino
```

Comment:
```
Tring can read certificates and private keys encoded in PEM encoded PKCS #1,  PEM encoded PKCS #8, PKCS #12, and the legacy Java KeyStore format.
```
The rest of the paragraph is not really accurate, so I'd drop it

Reply:
done